### PR TITLE
.github/workflows: Add check-csv-ordering.yaml

### PR DESCRIPTION
**Pull request by szabba**
Taking a stab at #9:
* Generates two versions of a file: one without the header and a sorted version of that.
* The pipeline fails when the file differs from what's expected.
    * This is the bareset-bones tool - it reminds people when things are wrong. It does nothing on it's own.
        * I hope this is easier to maintain long term.
    * You can see the pipeline both failing and succeeding in some of the runs [here](https://github.com/szabba/fate-product-list/actions/workflows/check-csv-ordering.yaml).
* The publisher name column is now first, followed by the product name.
* The CSV is reodered so it should not trigger the pipeline.

### CODE REVIEW

**Comment by szabba** [comment]
@amazingrando @johnhelmuth not sure if you'd need to configure something extra for the action to run on this PR.

File: .github/workflows/check-csv-ordering.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Check CSV order
+
+on:
+
+  push:
```
Comment (szabba) [inline .github/workflows/check-csv-ordering.yaml:5]: The pipeline didn't seem to run on my fork without this part. We'd want to remove it before merging.

File: .github/workflows/check-csv-ordering.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    branches:
+      - ci-check-csv-ordering
+
+  pull_request:
```
Comment (szabba) [inline .github/workflows/check-csv-ordering.yaml:9]: Not sure if this is correct and how that interacts with the source branch in the PR being on a fork.

File: .github/workflows/check-csv-ordering.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Generate header-free version
+        run: tail -n +2 fate-product-list.csv > headerless.csv
+
+      - name: Generate sorted version
+        run: cat headerless.csv | LC_ALL=en_US.UTF-8 sort | head -c -1 > sorted.csv
```
Comment (johnhelmuth) [inline .github/workflows/check-csv-ordering.yaml:25]: `sort` accepts a file to sort on the command line, so this should work:

```bash
LC_ALL=en_US.UTF-8 sort headerless.csv | head -c -1 > sorted.csv
```

File: .github/workflows/check-csv-ordering.yaml (hunk 1 above)
Comment (johnhelmuth) [reply .github/workflows/check-csv-ordering.yaml:5]: I got this to work by using the syntax:

```yaml
on: pull_request
```
 AND having the workflow file in the target branch of the PR.

(It also worked with a branches stanza, but I was curious what didn't work in or original so I tried various things, I think this is the simplest for this case.)

See PRs #3 through #5 in [my repo](https://github.com/johnhelmuth/fate-product-list/pulls?q=is%3Apr), and branches `test-ci-check-csv-ordering`, `test-ci-check-csv-ordering-should-fail`, and `test-ci-check-csv-ordering-should-succeed`.

Comment (johnhelmuth) [reply .github/workflows/check-csv-ordering.yaml:5]: I created a PR against your repo's branch with that configuration so you can decide if that's how you want it to work... merge that if you want, it should update this PR, then we can merge this one.

File: .github/workflows/check-csv-ordering.yaml (hunk 3 above)
Comment (szabba) [reply .github/workflows/check-csv-ordering.yaml]: Applied.

**Comment by johnhelmuth** [comment]
This looks good to me, tested it on my fork of the upstream repo.

**Pull request merged**